spec/lib/pact_broker/client/cli/broker_create_or_update_webhook_spec.rb

```diff
@@ -0,0 +1,221 @@
+              end
+              subject
+            end
+          end
+
+          context "when a uuid is provided" do
+            before do
+              options_hash.merge!(uuid: '9999')
+              expected_params.merge!(uuid: '9999')
+
+              broker.options = OpenStruct.new(options_hash)
+            end
+
+            it "calls PactBroker::Client::Webhooks::Create with uuid in params" do
+              expect(PactBroker::Client::Webhooks::Create).to receive(:call) do | params, _, _ |
+                expect(params).to eq expected_params
+                command_result
+              end
+              subject
+            end
+          end
+
+          context "when a file reference is passed for the data" do
+            before do
+              FileUtils.mkdir_p "tmp"
+              File.open("tmp/body.json", "w") { |file| file << "file" }
+            end
+
+            let(:data) { "@tmp/body.json" }
+
+            it "reads the file and passes it in to the Webhooks::Create call" do
+              expect(PactBroker::Client::Webhooks::Create).to receive(:call) do | params, _, _ |
+                expect(params[:body]).to eq "file"
+                command_result
+              end
+              subject
+            end
+
+            context "when the file is not found" do
+
+              let(:data) { "@doesnotexist.json" }
+
+              it "raises a PactBroker::Client::Error" do
+                expect { subject }.to raise_error PactBroker::Client::Error, /Couldn't read data from file/
+              end
+            end
+
+            context "when successful" do
+              it "prints the message to stdout" do
+                expect($stdout).to receive(:puts).with('message')
+                subject
+              end
+            end
+
+            context "when not successful" do
+              let(:success) { false }
+
+              it "prints the message to stderr" do
+                expect($stdout).to receive(:puts).with('message')
+                begin
+                  subject
+                rescue SystemExit
+                end
+              end
+
+              it "exits with code 1" do
+                exited_with_error = false
+                begin
+                  subject
+                rescue SystemExit
+                  exited_with_error = true
+                end
+                expect(exited_with_error).to be true
+              end
+            end
+
+            context "when a PactBroker::Client::Error is raised" do
+              before do
+                allow(PactBroker::Client::Webhooks::Create).to receive(:call).and_raise(PactBroker::Client::Error, "foo")
+              end
+
+              it "raises a WebhookCreationError which does not show an ugly stack trace" do
+                expect { subject }.to raise_error(WebhookCreationError, /foo/)
+              end
+            end
+          end
+        end
+      end
+    end
+  end
+end
```

Comment:
That makes sense, this isn't DRY code so I should fix this. However, I'm wondering if there's another way to do this:

At the moment, both commands (`create_webhook` and `create_or_update_webhook`) call a new method `run_webhook_commands`. Could I change the tests so that I have spec for the implementation of the `run_webhook_commands` method, and then have a smaller spec for each command ensuring they each call this method?

Re: the extra spec you mentioned for testing calling the `create_or_update_webhook` command without a uuid - do I need to test this if Thor already checks for the presence of a required parameter?

Reply:
> Could I change the tests so that I have spec for the implementation of the run_webhook_commands method, and then have a smaller spec for each command ensuring they each call this method?

Sure, give it a go.

> do I need to test this if Thor already checks for the presence of a required parameter?

I'm happy to skip it.